### PR TITLE
Update 10 minutes to cuDF

### LINKED ISSUE #4665: [DOC] Update 10 minutes to dask-cudf
### Report improvements to documentation

**Location of incorrect documentation**
[Link](https://rapidsai.github.io/projects/cudf/en/0.12.0/10min.html)

**Describe the problems or issues found in the documentation**
- `10 minutes to dask-cudf` should be refreshed (in 0.14). Here are a couple of suggestions, but it would be good to read through and update content wherever applicable.
  - Reduce the number of examples that perform `df.compute()` which will not work with larger datasets [(example)](https://github.com/rapidsai/cudf/blob/branch-0.13/docs/cudf/source/10min.ipynb?short_path=c1a961d#L5598).
  - Update `to_parquet` segment since we support GPU accelerated parquet writing.

### LINKED ISSUE #4946: [DOC] string docs for basic regex actions
## Report incorrect documentation

**Location of incorrect documentation**

https://docs.rapids.ai/api/cudf/stable/search.html?q=search&check_keywords=yes&area=default#

**Describe the problems or issues found in the documentation**

Searching docs for `search` and `match` had no hits for the new methods. I expected docs that would guide us on:

* String match
* Regex match
* How to get the first group out / NA
* How to get a list of matches out
* How to convert non-strings to strings and get matches out

**Steps taken to verify documentation is incorrect**

Slack

**Suggested fix for documentation**

1. API method docs

2. Include regex as part of the quick tutorials, esp. for the above actions of str/rgx match (bool/span), single/multiple groups (val), and non-str -> str.

---

### LINKED ISSUE #10614: [BUG] 10 minutes to cuDF and dask-cuDF notebook references missing data
**Describe the bug**
The `read_orc` section throw this error: ```FileNotFoundError: [Errno 2] No such file or directory: ['/rapids/cudf/python/cudf/cudf/tests/data/orc/TestOrcFile.test1.orc']```

**Steps/Code to reproduce bug**
Runt the notebook.

**Expected behavior**
Notebook should run.

### LINKED ISSUE #11931: [DOC] Update 10 minutes to cudf and dask-cudf for clarity on compute vs persist
[10 Minutes to cuDF and Dask-cuDF](https://docs.rapids.ai/api/cudf/nightly/user_guide/10min.html) is a core piece of the cuDF documentation and provides an excellent introduction to using cuDF and Dask.

However, this document heavily uses `ddf.compute()` because the data sizes are so small. The result is that many users end up calling `compute` and running out of memory, not realizing that they should have instead used `persist` or just viewed a portion of the first partition with `head`. 

We have [a section at the bottom](https://docs.rapids.ai/api/cudf/nightly/user_guide/10min.html#dask-performance-tips) that explains some of this in more depth, but it doesn't fully cover this distinction, many readers may not see it, and readers may have already internalized the pattern of `ddf.compute()`.

I'd propose that we:
- Add a very brief section describing compute vs persist vs head to the top of the documentation (linking to Dask docs or the relevant Coiled/Dask blogs as appropriate)
- Switch the use of `compute` to `head`.

Separately, we should also consider whether an update should include the Dask backend IO dispatching work  in https://github.com/rapidsai/cudf/pull/11920

### PR DESCRIPTION
**Pull request by wence-**
## Description

Update the introductory documentation to take account of current best practices.
In particular, use `compute` judiciously on `dask-cuDF` objects, and explain why
we are doing so.

Closes #4665
Closes #4946
Closes #10614
Closes #11931

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/cudf/source/user_guide/10min.ipynb [hunk 1]
```diff
@@ -2,20 +2,25 @@
  "cells": [
```
Comment (galipremsagar) [inline docs/cudf/source/user_guide/10min.ipynb:2]: <p>Could we manually edit the file to remove warnings &amp; full stack-trace that contains absolute paths of your machine?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/12114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (galipremsagar) [inline docs/cudf/source/user_guide/10min.ipynb:2]: <p>is the newline before <code>for more information</code> intentional? or a render issue on my end?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/12114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='77'/>

Comment (galipremsagar) [inline docs/cudf/source/user_guide/10min.ipynb:2]: <p>Can we edit the raw output here too? I think Bradley had a tool to clean up these paths.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/12114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='167'/>

Comment (galipremsagar) [inline docs/cudf/source/user_guide/10min.ipynb:2]: <p>Same as above about the absolute paths in nvidia-smi output.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/12114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='170'/>

Comment (galipremsagar) [inline docs/cudf/source/user_guide/10min.ipynb:2]: <pre class="ql-syntax" spellcheck="false">Line #1.    wait(results_ddf)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/12114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='177'/>

Comment (galipremsagar) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Screen Shot 2022-11-17 at 5 28 34 PM

Comment (bdice) [reply docs/cudf/source/user_guide/10min.ipynb:2]: https://github.com/bdice/nb-strip-paths

Must be run on the local machine before committing — CI won’t know what paths are local. I filed a SWIPAT but I’m not sure if it was completed or not.

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: I think this is an artifact of reviewnb rendering.

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Yes.

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Do we want to show the futures? I can do so, but maybe I should add some explanation.

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: <p>But I will check if it exists: it should not.</p>

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Artifact.

Comment (galipremsagar) [reply docs/cudf/source/user_guide/10min.ipynb:2]: <p>Not necessary here, My suggestion was to replace <code>wait(results_ddf);</code> with <code>wait(results_ddf)</code>   </p>

[the remark this answers is not in the document]
Comment (galipremsagar) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Yeah, we use `CudfEngine`: https://github.com/rapidsai/cudf/blob/branch-23.02/python/dask_cudf/dask_cudf/io/parquet.py#L468

[the remark this answers is not in the document]
Comment (galipremsagar) [reply docs/cudf/source/user_guide/10min.ipynb:2]: cc: @rjzamora for vis

[the remark this answers is not in the document]
Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Fixed.

Comment (wence-) [reply docs/cudf/source/user_guide/10min.ipynb:2]: Ah, the semicolon is the thing that suppresses the output. I just removed it and put the `repr` of the return value in.

**Comment by wence-** [comment]
Think I addressed all the comments.

**Review by galipremsagar — APPROVED** [review]
Thanks @wence- for updating the notebook!

File: docs/cudf/source/user_guide/10min.ipynb [hunk 2]
```diff
@@ -496,22 +417,24 @@
...
   },
   {
    "cell_type": "markdown",
+   "id": "0573b7bb",
    "metadata": {},
    "source": [
     "Creating a `cudf.DataFrame` from a pandas `Dataframe` and a `dask_cudf.Dataframe` from a `cudf.Dataframe`.\n",
     "\n",
-    "*Note that best practice for using Dask-cuDF is to read data directly into a `dask_cudf.DataFrame` with something like `read_csv` (discussed below).*"
+    "*Note that best practice for using dask-cuDF is to read data directly into a `dask_cudf.DataFrame` with `read_csv` other other builtin I/O routines (discussed below).*"
```
Comment (shwina) [inline docs/cudf/source/user_guide/10min.ipynb:431]:
```suggestion
    "*Note that best practice for using dask-cuDF is to read data directly into a `dask_cudf.DataFrame` with `read_csv` or other builtin I/O routines (discussed below).*"
```

**Pull request merged**